Packs/CyberArkEPM/Integrations/CyberArkEPMEventCollector/CyberArkEPMEventCollector_test.py

```diff
@@ -0,0 +1,262 @@
+                       json=mock_response_no_more_events)
+    requests_mock.post('https://mock.com/EPM/API/Sets/id2/policyaudits/search?nextCursor=start&limit=10',
+                       json=mock_response_policy_audits)
+    requests_mock.post('https://mock.com/EPM/API/Sets/id2/policyaudits/search?nextCursor=1700097106000&limit=10',
+                       json=mock_response_no_more_events)
+    requests_mock.post('https://mock.com/EPM/API/Sets/id1/Events/Search?nextCursor=start&limit=10', json=mock_response_events)
+    requests_mock.post('https://mock.com/EPM/API/Sets/id1/Events/Search?nextCursor=1702360757618&limit=10',
+                       json=mock_response_no_more_events)
+    requests_mock.post('https://mock.com/EPM/API/Sets/id2/Events/Search?nextCursor=start&limit=10', json=mock_response_events)
+    requests_mock.post('https://mock.com/EPM/API/Sets/id2/Events/Search?nextCursor=1702360757618&limit=10',
+                       json=mock_response_no_more_events)
+
+    return Client(
+        'https://url.com',
+        'test',
+        '123456',
+        '1',
+    )
+
+
+""" TEST HELPER FUNCTION """
+
+
+def test_create_last_run():
+    """
+        Given:
+            - A list of set_ids.
+
+        When:
+            - create_last_run function is running.
+
+        Then:
+            - Validates that the function works as expected.
+    """
+    from CyberArkEPMEventCollector import create_last_run
+
+    set_ids = ['123', '456']
+    from_date = '2023-01-01T00:00:00Z'
+    expected_result = {
+        '123': {
+            'admin_audits': {'from_date': from_date},
+            'policy_audits': {'from_date': from_date, 'next_cursor': 'start'},
+            'detailed_events': {'from_date': from_date, 'next_cursor': 'start'},
+        },
+        '456': {
+            'admin_audits': {'from_date': from_date},
+            'policy_audits': {'from_date': from_date, 'next_cursor': 'start'},
+            'detailed_events': {'from_date': from_date, 'next_cursor': 'start'},
+        }
+    }
+
+    assert create_last_run(set_ids, from_date) == expected_result
+
+
+def test_prepare_datetime():
+    """
+        Given:
+            - A datetime presentation
+                1. in str
+                2. in datetime object
+
+        When:
+            - prepare_datetime function is running.
+                1. with increase set to false.
+                2. with increase set to true.
+
+        Then:
+            - Validates that the function works as expected.
+    """
+    from CyberArkEPMEventCollector import prepare_datetime
+
+    str_from_date = '2023-01-01T00:00:00'
+    datetime_from_date = datetime.datetime.strptime(str_from_date, '%Y-%m-%dT%H:%M:%S')
+    expected_from_date = '2023-01-01T00:00:00.000Z'
+    expected_increase_from_date = '2023-01-01T00:00:00.001Z'
+
+    assert prepare_datetime(datetime_from_date) == expected_from_date
```

Comment:
Usually we use `@pytest.mark.parametrize`
To test different cases in the same unit test